filesystem/resource_filesystem_file_test.go

```diff
@@ -0,0 +1,156 @@
+package filesystem
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"syscall"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccFilesystemFile(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		Providers: map[string]terraform.ResourceProvider{"filesystem": Provider()},
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Check: resource.ComposeAggregateTestCheckFunc(testFilesystemFileCreate),
+				Config: `
+resource "filesystem_file" "test" {
+  path = "/tmp/testfile"
+  content = "blah"
+  mode = "0600"
+}
+`,
+			},
+			resource.TestStep{
+				Check: resource.ComposeAggregateTestCheckFunc(testFilesystemFileUpdateMode),
+				Config: `
+resource "filesystem_file" "test" {
+  path = "/tmp/testfile"
+  content = "blah"
+  mode = "0644"
+}
+`,
+			},
+			resource.TestStep{
+				Check: resource.ComposeAggregateTestCheckFunc(testFilesystemFileUpdateContent),
+				Config: `
+resource "filesystem_file" "test" {
+  path = "/tmp/testfile"
+  content = "yay"
+  mode = "0644"
+}
+`,
+			},
+		},
+		CheckDestroy: testFilesystemFileDelete,
+	})
+}
+
+func testFilesystemFileCreate(state *terraform.State) error {
```

Comment:
I would probably split this function in several one in order to allow reusability. For instance, having:
- 1 function for checking that the path exists
- 1 function for checking the user id
- 1 function for checking the username
- etc...

Thoughts?

Then line 40 (for example) could be rewritten as:
```go
			resource.TestStep{
				Check: resource.ComposeAggregateTestCheckFunc(
					testCheckFilesystemPathExists(),
					testCheckFilesystemId(),
					...,
				),
```